libraries/ESP8266WiFi/src/ESP8266WiFiGeneric.cpp

```diff
@@ -426,6 +432,65 @@ bool ESP8266WiFiGenericClass::forceSleepWake() {
     return false;
 }
 
+/**
+ * Get listen interval of maximum sleep level for modem sleep and light sleep.
+ * @return interval
+ */
+uint8_t ESP8266WiFiGenericClass::getListenInterval () {
+    return wifi_get_listen_interval();
+}
+
+/**
+ * Set listen interval of maximum sleep level for modem sleep and light sleep.
+ * PREFER using setSleepMode()
+ * datasheet:
+   It only works when sleep level is set as MAX_SLEEP_T
+   It should be called following the order:
+     wifi_set_sleep_level(MAX_SLEEP_T)
+     wifi_set_listen_interval
+     wifi_set_sleep_type
+   forum: https://github.com/espressif/ESP8266_NONOS_SDK/issues/165#issuecomment-416121920
+   default value seems to be 3 (as recommended by https://routerguide.net/dtim-interval-period-best-setting/)
+ */
+void ESP8266WiFiGenericClass::setListenInterval (uint8_t listenInterval) {
+    wifi_set_listen_interval(listenInterval);
+}
+
+/**
+ * Get sleep level of modem sleep and light sleep
+ * @return true if max level
+ */
+bool ESP8266WiFiGenericClass::isSleepLevelMax () {
+    return wifi_get_sleep_level() == MAX_SLEEP_T;
+}
+
+/**
+ * Set sleep level of modem sleep and light sleep
+ * PREFER using setSleepMode()
+ * datasheet:
+
+   This configuration should be called before calling wifi_set_sleep_type
+
+   Modem-sleep and light sleep mode have minimum and maximum sleep levels.
+
+   - In minimum sleep level, station wakes up at every DTIM to receive
+     beacon.  Broadcast data will not be lost because it is transmitted after
+     DTIM.  However, it can not save much more power if DTIM period is short,
+     as specified in AP.
+
+   - In maximum sleep level, station wakes up at every listen interval to
+     receive beacon.  Broadcast data may be lost because station may be in sleep
+     state at DTIM time.  If listen interval is longer, more power will be saved, but
+     it’s very likely to lose more broadcast data.
+
+   - Default setting is minimum sleep level.
+
+   Further reading: https://routerguide.net/dtim-interval-period-best-setting/
+
+ */
+void ESP8266WiFiGenericClass::setSleepLevelMax (bool setToMax) {

```

Comment:
Same here, sdk docs say this is part of a call sequence, so exposing it stand alone allows the user wrong usage.